### PR TITLE
OpenXRVk: Use PAL to initialize xr loader on Android

### PR DESCRIPTION
**Pull request by moraaar**
Signed-off-by: moraaar <moraaar@amazon.com>

### CODE REVIEW

File: Gems/OpenXRVk/Code/Source/OpenXRVkInstance.cpp [hunk 1]
```diff
@@ -88,26 +85,11 @@ namespace OpenXRVk
 
     AZ::RHI::ResultCode Instance::InitInstanceInternal(AZ::RHI::ValidationMode validationMode)
     {
-#ifdef AZ_PLATFORM_ANDROID
-        // Initialize the loader for Android platform
+#if OPENXRVK_NEEDS_INITIALIZE_XR_LOADER
```
Comment (amzn-phist) [inline Gems/OpenXRVk/Code/Source/OpenXRVkInstance.cpp:88]: For PAL defines like this it's common to use the word `TRAIT` in the name (e.g. `OPENXRVK_TRAIT_NEEDS_INITIALIZE_XR_LOADER`).

File: Gems/OpenXRVk/Code/Source/Platform/Android/OpenXRInitializeLoader_Android.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Android/AndroidEnv.h>
+
+#include <OpenXRVk/OpenXRVkUtils.h>
+#include <OpenXRInitializeLoader_Android.h>
+
+namespace OpenXRVk
```
Comment (amzn-phist) [inline Gems/OpenXRVk/Code/Source/Platform/Android/OpenXRInitializeLoader_Android.cpp:14]: Consider naming this `namespace OpenXRVk::Platform`, it gives a good indication at call site that there's platform-specific implementation behind it.

**Review by amzn-phist — APPROVED** [review]

File: Gems/OpenXRVk/Code/Source/OpenXRVkInstance.cpp (hunk 1 above)
Comment (moraaar) [reply Gems/OpenXRVk/Code/Source/OpenXRVkInstance.cpp:88]: Excellent point. I'll change it.

File: Gems/OpenXRVk/Code/Source/OpenXRVkInstance.cpp [hunk 3]
```diff
@@ -88,26 +85,11 @@ namespace OpenXRVk
 
     AZ::RHI::ResultCode Instance::InitInstanceInternal(AZ::RHI::ValidationMode validationMode)
     {
-#ifdef AZ_PLATFORM_ANDROID
-        // Initialize the loader for Android platform
+#if OPENXRVK_TRAIT_NEEDS_INITIALIZE_XR_LOADER
```
Comment (moudgils) [inline Gems/OpenXRVk/Code/Source/OpenXRVkInstance.cpp:88]: Do we need this macro if we have a common implementation of OpenXRInitializeLoader that returns true. Makes the code a little messy.

Comment (moraaar) [reply Gems/OpenXRVk/Code/Source/OpenXRVkInstance.cpp:88]: I think it's a larger change doing it that way, plus adding a cpp and a header just to do an empty function seems like a waste, this way it's just a one line macro.

**Review by moudgils — APPROVED** [review]

**Pull request merged**